graphbolt/src/csc_sampling_graph.cc

```diff
@@ -612,6 +614,11 @@ inline torch::Tensor LaborPick(
     const torch::TensorOptions& options,
     const torch::optional<torch::Tensor>& probs_or_mask,
     SamplerArgs<SamplerType::LABOR> args) {
+  // TODO: fix inconsistency with Neighbor sampler.
+  // 1. Replace = true, fanout = -1. Expected: sample all neighbors with
+  // non-zero probility once regardless of replacement.
+  // 2. Replace = true, fanout > num_neighbors. Expected: sample fanout many
+  // neighbors.
```

Comment:
@mfbalin Could you please help modify Labor sampler? Since you have the best knowledge of this process. :)

Reply:
Why don't we just call NeighborSampler if fanout is -1 and replace is true? I don't think replicating code makes a lot of sense. We can do this during dispatch. When fanout is -1, it doesn't matter whether labor or neighbor is used.

Reply:
I think the change should look like after we ensure that labor is never called with `fanout == -1`:
`fanout = replace ? fanout : std::min(fanout, num_neighbors);`

Reply:
And inside the `Pick<SamplerType::LABOR>` function, we can call `NonuniformPick` and `UniformPick` if `fanout < 0`. This way, there won't be duplicated code and will be a single defined behavior.

Reply:
See this commit: https://github.com/mfbalin/dgl-1/commit/5c3f8de64e1a09447be05868a85a7998b279ae7e